### PR TITLE
Models

### PR DESCRIPTION
**Pull request by AlexanderHHansen**
Made the code to create different models. For the regression model, a "snap to closest speed limit" function has been implemented to be used for those. Functionality for scoring our models is implemented. I used metrics typically used in regression models, as I think those metrics are more appropriate than those for classification. Much of the code is not really typed, because it wraps code from sklearn, which does not support type hints...

### CODE REVIEW

File: pipeline/models/models.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    return grid_search.best_estimator_, grid_search.best_params_  # type: ignore
+
+
+def statistical_model() -> StatisticalModel:
+    """Create a basic statistical model
+
+    Returns:
+        StatisticalModel: a statistical model with a predict function
+    """
+    return StatisticalModel()
```
Comment (anesheim-uni) [inline pipeline/models/models.py:177]: Does this work?

**Review by anesheim-uni — APPROVED** [review]
Virkelig godt arbejde!

File: pipeline/models/utils/scoring.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    Returns closest value in speed_limits.
+    If two numbers are equally close, return the smallest number.
+    """
+    pos = bisect_left(SPEED_LIMITS, speed)
+    if pos == 0:
+        return SPEED_LIMITS[0]
+    if pos == len(SPEED_LIMITS):
+        return SPEED_LIMITS[-1]
+    before = SPEED_LIMITS[pos - 1]
+    after = SPEED_LIMITS[pos]
```
Comment (JakobFaarGreg) [inline pipeline/models/utils/scoring.py:20]:
```suggestion
    index = bisect_left(SPEED_LIMITS, speed)
    if index == 0:
        return SPEED_LIMITS[0]
    if index == len(SPEED_LIMITS):
        return SPEED_LIMITS[-1]
    before = SPEED_LIMITS[index - 1]
    after = SPEED_LIMITS[index]
```

File: tests/models/scoring/test_scoring.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import pytest
+from pipeline.models.utils.scoring import (
+    find_closest_speed_limit,
+    mean_absolute_percentage_error,
+)
+
+# SPEED_LIMITS = [15,30,40,50,60,70,80,90,100,110,120,130]
+
```
Comment (JakobFaarGreg) [inline tests/models/scoring/test_scoring.py:8]:
```suggestion
```

File: pipeline/models/utils/scoring.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import pandas as pd  # type: ignore
+from bisect import bisect_left
+
+SPEED_LIMITS = [15, 30, 40, 50, 60, 70, 80, 90, 100, 110, 120, 130]
+
+
+def find_closest_speed_limit(speed: float) -> float:
+    """
+    Returns closest value in speed_limits.
+    If two numbers are equally close, return the smallest number.
```
Comment (JakobFaarGreg) [inline pipeline/models/utils/scoring.py:12]: Tilføj grunden dertil, det føles arbitrært som det er skrevet nu. 
Men det er grundlæggende fordi det er sikkerhedsmæssigt bedre at køre langsommere og man kommer ikke i bødeproblemer ift. at køre for stærkt

**Review by JakobFaarGreg — CHANGES_REQUESTED** [review]

**Review by JakobFaarGreg — APPROVED** [review]

**Pull request merged**